### ISSUE TITLE
Bug: parser can't resolve imports when a package is using `typesVersions`

### ISSUE THREAD

**Issue by zikaari** [typescript-eslint/typescript-eslint#5241]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Relevant Package

parser

### Repro Code

```typescript
import {} from '@mozilla/glean/webext'
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  "parserOptions": {
    "ecmaVersion": 12,
    "sourceType": "module"
  },
  rules: {
    "@typescript-eslint/<rule-name>": ["error", ...<options>],
  },
  "settings": {
    "import/parsers": {
      "@typescript-eslint/parser": [".ts", ".tsx"]
    },
    "import/resolver": {
      "typescript": {
        "alwaysTryTypes": true,
        "project": ["./tsconfig.eslint.json", "./packages/*/tsconfig.json"]
      }
    },
    "react": {
      "version": "18"
    }
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "module": "esnext",
    "target": "esnext",
    "moduleResolution": "node",
    "allowSyntheticDefaultImports": true,
    "esModuleInterop": true,
    "declaration": true,
    "baseUrl": ".",
    "types": ["@types/node"],
    "noEmit": true,
    "allowJs": true
  }
}
```

### Expected Result

The import should be resolved (the TS parser itself is having no problems locating the module, and VSCode's intellisense is able to suggest exports from that import)
![parser](https://user-images.githubusercontent.com/11316020/175750871-9cb30ea1-d794-4d22-9c0b-a960f55fcffd.png)

### Actual Result

Unable to resolve the import

### Additional Info

See `typesVersions` of the package in question - https://github.com/mozilla/glean.js/blob/main/glean/package.json#L19-L49

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `^5.25.0` |
| `@typescript-eslint/parser`            | `^5.25.0` |
| `@typescript-eslint/scope-manager`     | `X.Y.Z` |
| `@typescript-eslint/typescript-estree` | `X.Y.Z` |
| `@typescript-eslint/type-utils`        | `X.Y.Z` |
| `@typescript-eslint/utils`             | `X.Y.Z` |
| `TypeScript`                           | `^4.6.4` |
| `ESLint`                               | `^8.15.0` |
| `node`                                 | `v18.3.0` |

**Comment by bradzacher** [comment]
Hey @zikaari - this is a bug with another project.

Please file an issue with them! 

https://github.com/alexgorbatchev/eslint-import-resolver-typescript
